### PR TITLE
working onLifting State

### PR DESCRIPTION
**Pull request by tadesse123**
Controlled Components
Props handling

### CODE REVIEW

**Review by kaytiwiita — COMMENTED** [review]
Looking really good! nice job passing in destructured variables as props. web app is really starting to come together! Still just some work on indentation, but functionally working really well.

File: src/AddTodoForm.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
 import React from 'react';
 
-function AddTodoForm(props){
+function AddTodoForm({onAddTodo}){
+    const[todoTitle, setTodoTitle]=React.useState("");
```
Comment (kaytiwiita) [inline src/AddTodoForm.js:4]: keep your spacing consistent. should be written out as const [todoTitle, setTodoTitle] = useState("").

File: src/AddTodoForm.js [hunk 2]
```diff
@@ -1,19 +1,25 @@
 import React from 'react';
```
Comment (kaytiwiita) [inline src/AddTodoForm.js:1]: can import {useState} from the react library as well so you don't need to reference React as the parent library when using it in line 4. import statement would look like: 

import React, { useState } from 'react';

File: src/AddTodoForm.js [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+    };
+
     const handleAddTodo = (event)=>{
         event.preventDefault();
 
-        const todoTitle = event.target.title.value;
-
-        props.onAddTodo(todoTitle);
+       
+        onAddTodo({title :todoTitle ,id:Date.now()});
```
Comment (kaytiwiita) [inline src/AddTodoForm.js:15]: keep your spacing consistent. {title: todoTitle, id: Date.now()}

File: src/AddTodoForm.js [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
 
-        event.target.reset();
+        setTodoTitle("")
     };
     return (
             <form onSubmit={handleAddTodo}>
                 <label htmlFor ="TodoTitle">Title</label>
-                <input type="text" name="title" id='todoTitle'></input>
+                <input onChange = {handleTitleChange} value ={todoTitle} type="text" name="title" id="todoTitle"></input>
```
Comment (kaytiwiita) [inline src/AddTodoForm.js:22]: can use self-closing tag on input

File: src/App.js [hunk 5]
```diff
@@ -1,15 +1,19 @@
 import React from 'react';
+
 import TodoList from "./TodoList";
 import AddTodoform from "./AddTodoForm";
 function App() {
-  const [newTodo, setNewTodo] = React.useState("")
+  const[todoList, setTodoList] = React.useState([]);
```
Comment (kaytiwiita) [inline src/App.js:6]: same as above, can import useState from React library so you don't need to reference it.

**Comment by tadesse123** [comment]
Thank for your feedback,I will check my code again and I will try to fix it.

**Pull request merged**